src/Plugin/search_api/processor/EntityReferenceWithUri.php

```diff
@@ -0,0 +1,195 @@
+
+use Drupal\islandora\Plugin\search_api\processor\Property\EntityReferenceWithUriProperty;
+use Drupal\search_api\Datasource\DatasourceInterface;
+use Drupal\search_api\Item\ItemInterface;
+use Drupal\search_api\Processor\ProcessorPluginBase;
+use Drupal\islandora\IslandoraUtils;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Add a filterable search api field for entity reference fields.
+ *
+ * @SearchApiProcessor(
+ *   id = "entity_reference_with_term",
+ *   label = @Translation("Entity Reference, where the target has a given taxonomy term"),
+ *   description = @Translation("Indexes the titles of Entity Reference field targets, where the targets have a taxonomy term or terms. Does not apply to Taxonomy Reference fields."),
+ *   stages = {
+ *     "add_properties" = 0,
+ *   },
+ *   locked = false,
+ *   hidden = false,
+ * )
+ */
+class EntityReferenceWithUri extends ProcessorPluginBase {
+
+  /**
+   * Islandora utils.
+   *
+   * @var \Drupal\islandora\IslandoraUtils
+   */
+  protected IslandoraUtils $utils;
+
+  /**
+   * Constructor.
+   *
+   * @param array $configuration
+   *   The plugin configuration, i.e. an array with configuration values keyed
+   *   by configuration option name. The special key 'context' may be used to
+   *   initialize the defined contexts by setting it to an array of context
+   *   values keyed by context names.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\islandora\IslandoraUtils $utils
+   *   Islandora utils.
+   */
+  public function __construct(
+    array $configuration,
+          $plugin_id,
+          $plugin_definition,
+    IslandoraUtils $utils
+  ) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->utils = $utils;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('islandora.utils'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getPropertyDefinitions(DatasourceInterface $datasource = NULL) {
+    $properties = [];
+    if (!$datasource || !$datasource->getEntityTypeId()) {
+      return $properties;
+    }
+
+    $entity_type = $datasource->getEntityTypeId();
+
+    // Get all configured Entity Relation fields for this entity type.
+    $fields = \Drupal::entityTypeManager()->getStorage('field_config')->loadByProperties([
+      'entity_type' => $entity_type,
+      'field_type' => 'entity_reference',
+    ]);
+
+    foreach ($fields as $field) {
+      // Skip over fields that point to taxonomy term fields themselves.
+      if ($field->getSetting('target_type') == 'taxonomy_term') {
+        continue;
+      }
+
+      $definition = [
```

Comment:
Would it potentially make sense to store the original field and bundle within the definition such that later on the original field name doesn't need to be reconstructed and can simplify the logic below?

Reply:
Yes. I also skipped the bundle, to better match Search API fields.